src/SourceBuild/tarball/content/tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CheckForPoison.cs

```diff
@@ -184,7 +182,7 @@ internal IEnumerable<PoisonedFileEntry> GetPoisonedFiles(IEnumerable<string> ini
                 // add its contents to the list to be checked.
                 if (ZipFileExtensions.Concat(TarFileExtensions).Concat(TarGzFileExtensions).Any(e => checking.ToLowerInvariant().EndsWith(e)))
                 {
-                    var tempCheckingDir = Path.Combine(tempDir.FullName, Path.GetRandomFileName(), Path.GetFileNameWithoutExtension(checking) + "." + (++dirCounter).ToString());
+                    var tempCheckingDir = Path.Combine(tempDir.FullName, Path.GetRandomFileName(), Path.GetFileNameWithoutExtension(checking));
```

Comment:
Can we get rid of the `Path.GetRandomFileName()` here?  It doesn't seem like it is needed.  If removed, the stripping logic in CheckSingleFile wouldn't be needed.

Reply:
Yes, I think you're right.  Removed this and the path stripping.  Thanks!